ext/opentelemetry-ext-asyncpg/tests/test_asyncpg_wrapper.py

```diff
@@ -0,0 +1,374 @@
+        self.assertEqual(
+            spans[0].attributes,
+            {"db.type": "sql", "db.statement.parameters": (1, 2, 3)},
+        )
+
+
+class TestAsyncPGInstrumentation(TestBase):
+    @mock.patch(
+        "asyncpg.Connection._do_execute",
+        new_callable=_coroutine_mock,
+        return_value=_do_execute_return_value,
+    )
+    def test_instrumented_connection(self, *_, **__):
+        AsyncPGInstrumentor().instrument()
+        connection = _get_connection()
+        _await(connection.execute("SELECT 42;", 1, 2, 3))
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 1)
+        self.assertEqual(
+            spans[0].attributes,
+            {
+                "db.type": "sql",
+                "db.statement.parameters": (1, 2, 3),
+                "db.statement": "SELECT 42;",
+            },
+        )
+
+    @mock.patch(
+        "asyncpg.Connection._do_execute",
+        new_callable=_coroutine_mock,
+        return_value=_do_execute_return_value,
+    )
+    def test_uninstrumented_connection(self, *_, **__):
+        AsyncPGInstrumentor().uninstrument()
+        self.memory_exporter.clear()
+        connection = _get_connection()
+        _await(connection.execute("SELECT 42;", 1, 2, 3))
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 0)
+
+    def test_instrumentation_flags(self):
+        AsyncPGInstrumentor().instrument()
+        for method_name in ["_execute", "_executemany"]:
+            method = getattr(Connection, method_name, None)
+            self.assertTrue(
+                hasattr(method, "_opentelemetry_ext_asyncpg_applied")
+            )
+        AsyncPGInstrumentor().uninstrument()
+        for method_name in ["_execute", "_executemany"]:
+            method = getattr(Connection, method_name, None)
+            self.assertFalse(
+                hasattr(method, "_opentelemetry_ext_asyncpg_applied")
+            )
+
+    def test_duplicated_instrumentation(self):
+        AsyncPGInstrumentor().instrument()
+        AsyncPGInstrumentor().instrument()
+        AsyncPGInstrumentor().instrument()
+        AsyncPGInstrumentor().uninstrument()
+        for method_name in ["_execute", "_executemany"]:
+            method = getattr(Connection, method_name, None)
+            self.assertFalse(
+                hasattr(method, "_opentelemetry_ext_asyncpg_applied")
+            )
+
+    def test_duplicated_uninstrumentation(self):
+        AsyncPGInstrumentor().instrument()
+        AsyncPGInstrumentor().uninstrument()
+        AsyncPGInstrumentor().uninstrument()
+        AsyncPGInstrumentor().uninstrument()
+        for method_name in ["_execute", "_executemany"]:
+            method = getattr(Connection, method_name, None)
+            self.assertFalse(
+                hasattr(method, "_opentelemetry_ext_asyncpg_applied")
+            )
+
+
+class TestAsyncPGConnectionMethods(TestBase):
```

Comment:
I assume none of these tests actually test on a postgres db? Should we add asyncpg tests to `docker-tests`? A postgres DB is already set up so it shouldn't be too hard to add tests for this

Reply:
agreed, new integration tests was deployed